python/paddle/fluid/distribute_transpiler.py

```diff
@@ -1147,3 +1140,25 @@ def _get_lr_ops(self):
                     # we only need to append op for once
                     break
         return lr_ops
+
+    def _get_optimize_pass(self):
+        block = self.origin_program.global_block()
+        opt_ops = []
+        params_grads = []
+        for op in block.ops:
+            if self._is_opt_op(op):
+                opt_ops.append(op)
+                params_grads.append((self.origin_program.global_block().var(
+                    op.input("Param")[0]),
+                                     self.origin_program.global_block().var(
+                                         op.input("Grad")[0])))
+            elif op.type == "scale":
```

Comment:
Can you use `ufind` here to find the connected ops so that we can remove this hack here?

Reply:
we only need `scale` op here, but `ufind` will find too many ops such as fc and fc_grad.

Reply:
I saw `get_pserver_program` also have similar code, can you put this in a private function?

Reply:
Done.